codec-http/src/test/java/io/netty/handler/codec/http/HttpResponseDecoderTest.java

```diff
@@ -146,6 +146,67 @@ public void testResponseChunked() {
         assertNull(ch.readInbound());
     }
 
+    @Test
+    public void testResponseDisallowPartialChunks() {
+        HttpResponseDecoder decoder = new HttpResponseDecoder(
+            HttpObjectDecoder.DEFAULT_MAX_INITIAL_LINE_LENGTH,
+            HttpObjectDecoder.DEFAULT_MAX_HEADER_SIZE,
+            HttpObjectDecoder.DEFAULT_MAX_CHUNK_SIZE,
+            HttpObjectDecoder.DEFAULT_VALIDATE_HEADERS,
+            HttpObjectDecoder.DEFAULT_INITIAL_BUFFER_SIZE,
+            HttpObjectDecoder.DEFAULT_ALLOW_DUPLICATE_CONTENT_LENGTHS,
+            false);
+        EmbeddedChannel ch = new EmbeddedChannel(decoder);
+
+        String headers = "HTTP/1.1 200 OK\r\n"
+            + "Transfer-Encoding: chunked\r\n"
+            + "\r\n";
+        ch.writeInbound(Unpooled.copiedBuffer(headers, CharsetUtil.US_ASCII));
+
+        HttpResponse res = ch.readInbound();
+        assertThat(res.protocolVersion(), sameInstance(HttpVersion.HTTP_1_1));
+        assertThat(res.status(), is(HttpResponseStatus.OK));
+
+        byte[] chunk = new byte[10];
+        for (int i = 0; i < chunk.length; i++) {
+            chunk[i] = (byte) i;
+        }
+
+        byte[] partialChunk1 = new byte[5];
+        System.arraycopy(chunk, 0, partialChunk1, 0, 5);
+        byte[] partialChunk2 = new byte[5];
+        System.arraycopy(chunk, 5, partialChunk2, 0, 5);
+
+        for (int i = 0; i < 10; i++) {
```

Comment:
Why do we have to write these chunks `10` times when the 2 `partialChunk`s contain all the data?

Reply:
Removed the for loop, no need to write these chunks 10 times